hudi-client/hudi-java-client/src/main/java/org/apache/hudi/table/action/commit/JavaUpsertPartitioner.java

```diff
@@ -205,6 +205,11 @@ private void assignInserts(WorkloadProfile profile, HoodieEngineContext context)
 
   private Map<String, List<SmallFile>> getSmallFilesForPartitions(List<String> partitionPaths, HoodieEngineContext context) {
     Map<String, List<SmallFile>> partitionSmallFilesMap = new HashMap<>();
+
+    if (config.getParquetSmallFileLimit() == 0) {
```

Comment:
How about using <= 0 to indicate always writing to new file group? And maybe add a description of this behavior to the document of `PARQUET_SMALL_FILE_LIMIT`.

Reply:
Can `PARQUET_SMALL_FILE_LIMIT` be set to be negative? I'm not sure.

Yeah, I'll add the description to the `PARQUET_SMALL_FILE_LIMIT`

Reply:
It is possible to be set it to negative, which is meaningless of course. So I guess we could cover it as well?

Reply:
done